### PR TITLE
Modernize codebase with Clang-Tidy range based loops, part 1

### PR DESCRIPTION
**Pull request by Lillecarl**
**Changes proposed:**
-  Range based loops everywhere

**Target branch(es):**
- [x] 3.3.5
- [ ] master

**Issues addressed:** 
Keeping TrinityCore up2date with the C++ standards

**Tests performed:**
Builds locally with clang

**Known issues and TODO list:** (add/remove lines as needed)
- [ ] Finish the rest

### CODE REVIEW

File: src/server/game/AI/CoreAI/CombatAI.cpp
Comment (ratkosrb) [inline src/server/game/AI/CoreAI/CombatAI.cpp]: I don't think the iterator should have m_ in the name, since that stands for member.

Comment (MaxtorCoder) [reply src/server/game/AI/CoreAI/CombatAI.cpp]: It indeed should be `l_Spell` 😄 or just `spell` since Shauren hates this code style.

Comment (Lillecarl) [reply src/server/game/AI/CoreAI/CombatAI.cpp]: True, I'll update.

File: src/common/Logging/Logger.cpp
Comment (jackpoz) [inline src/common/Logging/Logger.cpp]: should this have & too ?

File: src/common/Utilities/EventMap.cpp
Comment (jackpoz) [inline src/common/Utilities/EventMap.cpp]: should this have & too ?

File: src/common/Utilities/EventMap.cpp
Comment (jackpoz) [inline src/common/Utilities/EventMap.cpp]: should this have & too ?

Comment (Lillecarl) [reply src/common/Utilities/EventMap.cpp]: Yep, and uint32 const, all keys should be const to be able to get a reference

File: src/common/Utilities/EventMap.cpp
Comment (Lillecarl) [reply src/common/Utilities/EventMap.cpp]: Yes, and uint32 const

File: src/common/Logging/Logger.cpp
Comment (Lillecarl) [reply src/common/Logging/Logger.cpp]: Yep

**Comment by jackpoz** [comment]
I think it can be merged after you deal with the last comments

**Review by jackpoz — CHANGES_REQUESTED** [review]
Please fix the comments above

**Comment by Lillecarl** [comment]
Done :smile:

**Comment by Lillecarl** [comment]
Idk how you want the workflow with PRs to work, should i let you guys rebase and make the history nice, or how do you want me to do change request fixes in the future?

**Comment by illfated** [comment]
The final merge will be squashed, making it 1 commit to the TC target branch.

**Comment by jackpoz** [comment]
@Lillecarl adding a new commit on top of the existing ones makes it easier to see what changed since last review. As Github allows to easily squash before merging, it doesn't matter if a PR has 100 commits :)

**Comment by jackpoz** [comment]
Thanks for the PR :)

**Comment by Lillecarl** [comment]
Thanks for the review mate 😁

**Pull request merged**
